woodwork/type_sys/inference_functions.py

```diff
@@ -120,7 +120,8 @@ def __call__(self, series: pd.Series) -> bool:
         regex = self.get_regex()
 
         # Includes a check for object dtypes
-        if not pdtypes.is_string_dtype(series.dtype):
+        is_object = pdtypes.is_object_dtype(series.dtype)
+        if is_object or not pdtypes.is_string_dtype(series.dtype):
             return False
```

Comment:
`is_string_dtype` can evaluate an object without strings to be true. I'm assuming we strictly want string dtypes

```
>>> pd.api.types.is_string_dtype(pd.Series([(3, 2)]))
True
```